### ISSUE TITLE
[IMPROVEMENT] Make fsGroupPolicy aware in CSIDriver

### ISSUE THREAD

**Issue by innobead** [longhorn/longhorn#3762]
## Is your improvement request related to a feature? Please describe

For now (<= 1.2.4), the Longhorn CSI driver does not respect the recently introduced `fsGroupPolicy` in CSIDriverSpec, and it will cause some confusion about the fs permission when users deploy workloads using RWX volumes. 

Because right now 1.2.4 uses 1.18 kube API, there are no any specific settings for `fsGroupPolicy`. However, we should provide a capability to ensure the CSI driver respect that configuration to resolve the fs permission behavior inconsistent from RWO volumes. 

## Describe the solution you'd like

Update the kube API to support `fsGroupPolicy` in the Longhorn driver and allow users to set the behavior they want or even be consistent with RWO behavior. 

## Describe alternatives you've considered

N/A

## Additional context

- https://github.com/longhorn/longhorn/issues/3661#issuecomment-1073535353

**Comment by PhanLe1010** [comment]
The problem here is that kubelet doesn't change ownership and permission for RWX volume when a pod with pod.spec.securityContext.fsGroup starts.

We should check setting `fsGroupPolicy=File` to see if it can meet our requirement: respect the setting `pod.spec.securityContext.fsGroupChangePolicy` for both RWO and RWX which is:
* if `pod.spec.securityContext.fsGroupChangePolicy=OnRootMismatch`, it should not recursive change the volume ownership and permissions unless the root is mismatch. (if it doesn't meet this requirement, this cannot be a solution because there will be a big delay when mounting volume, see more at https://github.com/longhorn/longhorn/issues/2131#issuecomment-778897129)
* if `pod.spec.securityContext.fsGroupChangePolicy=always`, it should always change the volume ownership.

**Comment by cbandy** [comment]
> We should check setting `fsGroupPolicy=File` to see if it can meet our requirement: respect the setting `pod.spec.securityContext.fsGroupChangePolicy` for both RWO and RWX

Agreed. The analysis in #2131 seems to say that `fsGroupPolicy=File` ignores the Pod settings:

> > when `csiDriver.spec.fsGroupPolicy` is set to `None` or `File`, it will overwrite the setting for `pod.spec.securityContext.fsGroupChangePolicy`.

My understanding is that `fsGroupPolicy=File` forces kubelet to respect Pod settings:

- https://github.com/kubernetes/kubernetes/blob/v1.23.0/pkg/volume/csi/csi_mounter.go#L402-L404
- https://github.com/kubernetes/kubernetes/blob/v1.23.0/pkg/volume/csi/csi_mounter.go#L287
- https://github.com/kubernetes/kubernetes/blob/v1.23.0/pkg/volume/volume_linux.go#L53
- https://github.com/kubernetes/kubernetes/blob/v1.23.0/pkg/volume/volume_linux.go#L110
